ilastik/shell/gui/reportIssueDialog.py

```diff
@@ -0,0 +1,120 @@
+import pathlib
+import platform
+import re
+import webbrowser
+from functools import partial
+
+from PyQt5.QtCore import QUrl, Qt
+from PyQt5.QtGui import QDesktopServices
+
+from ilastik import __version__ as ilastik_version
+from ilastik import ilastik_logging
+from PyQt5.QtWidgets import QApplication, QDialog, QVBoxLayout, QLabel, QTextEdit, QPushButton, QHBoxLayout, QSizePolicy
+from urllib.parse import quote
+
+FORUM_URI = "https://forum.image.sc/tag/ilastik"
+TEAM_EMAIL = "team@ilastik.org"
+FILE_PATH_MASK = "(file path masked)"
+LOG_LENGTH_THRESHOLD = 5000  # Maximum we are happy to have people email or post in the forum
+
+
+def _mask_file_paths(text: str) -> str:
+    """Replace file paths with a placeholder for privacy."""
+    windows_pattern = r"""       # Windows path or file-URI with Windows path (could be with double-backslash)
+        [a-zA-Z]:[/\\]\\?        # Drive and first slash
+        (?!/)                    # Not immediately followed by another slash (avoid matching schema://)
+        ([^/\\\n]+[/\\]\\?)*     # Any number of directories
+        \S*
+    """
+    unix_pattern = r"""          # Unix path or other URI
+            /[^/\n]+/            # at least two slashes, but not double-slash
+            \S*
+    """
+    combined_pattern = re.compile(f"({windows_pattern}|{unix_pattern})", re.X)
+    masked = re.sub(combined_pattern, FILE_PATH_MASK, text)
+    return masked
+
+
+def _get_current_session_log_or_hint() -> str:
+    log_path = ilastik_logging.get_session_logfile_path()
+    if not log_path:
+        return 'Could not find session log file. Please use the "Open log folder" button to locate the log.'
+    with open(log_path, "r") as log_file:
+        log_text = log_file.read()
+    if len(log_text) > LOG_LENGTH_THRESHOLD:
+        return 'Session log is too long to include here. Please use the "Open log folder" button to locate the log.'
+    return log_text
+
+
+class ReportIssueDialog(QDialog):
```

Comment:
I'd prefer to have this dialog modal - without it, the user can have the dialog open, do more in ilastik and send off an outdated log...

Reply:
I chose to have it as a floating dialog because I wanted to be able to use the "show log folder" menu option. Now that there's a button for that here, I'd be ok making it a modal.